### ISSUE TITLE
Unable to edit entries

### ISSUE THREAD

**Issue by nyn44e** [android-password-store/Android-Password-Store#520]
I seem to be unable to edit existing entries. I can edit them alright but tapping the checkmark icon doesn't commit the changes. When I leave the specific entry my changes get lost.

Android Version 8.0.0
App Version 1.3.2

**Comment by MSF-Jarvis** [comment]
Can you try [this](https://github.com/zeapo/Android-Password-Store/blob/master/app/release/app-release.apk) build and see if the problem persists?

**Comment by nyn44e** [comment]
No change with that build.

**Comment by MSF-Jarvis** [comment]
That's unfortunate. Can you capture a logcat while going through the process of making changes and post it here? I'll try to debug it when I'm back home.

**Comment by nyn44e** [comment]
There you go. It seems that the app is unable to write to the location. I checked the apps permissions again and they're sure set to allow it (that'
[logcat_06-09-2019_11-08-12.txt](https://github.com/zeapo/Android-Password-Store/files/3269385/logcat_06-09-2019_11-08-12.txt)
![Screenshot_20190609-111654_Package installer](https://user-images.githubusercontent.com/51502732/59157280-477ca280-8aa8-11e9-912c-de91fc896cc2.jpg)

s what it says on the attached screenshot). Thank you for looking into it.

**Comment by gardenappl** [comment]
The bug still occurs in v1.3.3, with the same exception message:
```
02-03 18:29:49.772  2809  2809 E PgpActivity: An Exception occurred
02-03 18:29:49.772  2809  2809 E PgpActivity: java.io.IOException: File '/storage/emulated/0/Pass/social/discord.gpg' cannot be written to
02-03 18:29:49.772  2809  2809 E PgpActivity:   at org.apache.commons.io.FileUtils.openOutputStream(FileUtils.java:350)
02-03 18:29:49.772  2809  2809 E PgpActivity:   at org.apache.commons.io.FileUtils.openOutputStream(FileUtils.java:319)
02-03 18:29:49.772  2809  2809 E PgpActivity:   at com.zeapo.pwdstore.crypto.PgpActivity$encrypt$1.onReturn(PgpActivity.kt:448)
02-03 18:29:49.772  2809  2809 E PgpActivity:   at org.openintents.openpgp.util.OpenPgpApi$OpenPgpAsyncTask.onPostExecute(OpenPgpApi.java:318)
02-03 18:29:49.772  2809  2809 E PgpActivity:   at org.openintents.openpgp.util.OpenPgpApi$OpenPgpAsyncTask.onPostExecute(OpenPgpApi.java:299)
02-03 18:29:49.772  2809  2809 E PgpActivity:   at android.os.AsyncTask.finish(AsyncTask.java:695)
02-03 18:29:49.772  2809  2809 E PgpActivity:   at android.os.AsyncTask.access$600(AsyncTask.java:180)
02-03 18:29:49.772  2809  2809 E PgpActivity:   at android.os.AsyncTask$InternalHandler.handleMessage(AsyncTask.java:712)
02-03 18:29:49.772  2809  2809 E PgpActivity:   at android.os.Handler.dispatchMessage(Handler.java:106)
02-03 18:29:49.772  2809  2809 E PgpActivity:   at android.os.Looper.loop(Looper.java:193)
02-03 18:29:49.772  2809  2809 E PgpActivity:   at android.app.ActivityThread.main(ActivityThread.java:6718)
02-03 18:29:49.772  2809  2809 E PgpActivity:   at java.lang.reflect.Method.invoke(Native Method)
02-03 18:29:49.772  2809  2809 E PgpActivity:   at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:491)
02-03 18:29:49.772  2809  2809 E PgpActivity:   at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)

```
I press the "Done" checkmark button and the app just doesn't do anything, the file is not modified.
This might be because I'm storing my files in external storage (but I double checked and the app does have write permissions).

**I'm unable to add new entries either,** even though other apps have no problem writing to this directory (e.g. I managed to create a text file using a third-party text editor and manually encrypt it using OpenKeychain). So I assume this is not a problem with permissions.

It's possible to work around this but it's very inconvenient.

**Comment by msfjarvis** [comment]
> The bug still occurs in v1.3.3, with the same exception message:
> _snip_

1.3.3 is no longer the latest release, and the issue is fixed on [v1.4.0](https://github.com/android-password-store/Android-Password-Store/releases/tag/v1.4.0).

**Comment by gardenappl** [comment]
> 1.3.3 is no longer the latest release, and the issue is fixed on [v1.4.0](https://github.com/android-password-store/Android-Password-Store/releases/tag/v1.4.0).
That's good to know, I'm still using v1.3.3 because that's the latest version available on F-Droid.

But if the bug is fixed, then why is the issue still open? And there's no mention of it being fixed.

**Comment by msfjarvis** [comment]
> > 1.3.3 is no longer the latest release, and the issue is fixed on [v1.4.0](https://github.com/android-password-store/Android-Password-Store/releases/tag/v1.4.0).
> 
> That's good to know, I'm still using v1.3.3 because that's the latest version available on F-Droid.

F-Droid is unlikely to get further updates because the package name was updated and it's not a release channel I'm particularly interested in the upkeep of. Someone from the community can feel free to submit a new listing to continue receiving updates, or switch to using the snapshot builds that I provide [here](https://dl.msfjarvis.dev/APS).

> But if the bug is fixed, then why is the issue still open? And there's no mention of it being fixed.

This issue wasn't closed because the logs here were useless and the bug had to be reported again [here](https://github.com/android-password-store/Android-Password-Store/issues/600) to be actionable.

**Comment by gardenappl** [comment]
Thank you for the quick reply. I'll see if I can help with submitting to F-Droid.
